### ISSUE TITLE
lint: add checker for unnessesary blocks

### ISSUE THREAD

**Issue by fexolm** [go-critic/go-critic#174]
Warn on blocks without local variables.
**before:**
```go
a := 5
{
     fmt.Print(a)
}
```
**after:**
```go
a := 5
fmt.Print(a)
```

**Comment by cristaloleg** [comment]
What about cases when user want to separate some logic?
‘’’
mtx,Lock()
{
// do something 
}
mtx.Unlock()

**Comment by Quasilyte** [comment]
@cristaloleg, I think such users could disable this check and get on with it.
If this check is trivially implementable, I see no reason why we shouldn't include it.

**Comment by cristaloleg** [comment]
By presenting this example I wanted to show that it's too broad to warn users, but it's trivial - why not.
